### ISSUE TITLE
Failing to find cl-change-case-test after all tests passed

### ISSUE THREAD

**Issue by Hellseher** [rudolfochrist/cl-change-case#2]
Hi

While packing Guix I've got build issue with following backtrace:

```
Invoking sbcl: "/gnu/store/wyfbw52srslpf3cbyjzfxhd16n7ilk91-sbcl-2.1.0/bin/sbcl" "--non-interactive" "--eval" "(require :asdf)" "--eval" "(asdf:load-asd (truename \"/gnu/store/96w6s8fk2y41yl6amkbr8h3z4d3av1dg-sbcl-cl-change-case-0.1.0-1.5ceff2a/share/common-lisp/sbcl/cl-cha
nge-case/cl-change-case.asd\"))" "--eval" "(asdf:load-asd (truename \"/gnu/store/96w6s8fk2y41yl6amkbr8h3z4d3av1dg-sbcl-cl-change-case-0.1.0-1.5ceff2a/share/common-lisp/sbcl/cl-change-case/cl-change-case-test.asd\"))" "--eval" "(asdf:load-asd (truename \"/gnu/store/96w6s8fk2
y41yl6amkbr8h3z4d3av1dg-sbcl-cl-change-case-0.1.0-1.5ceff2a/share/common-lisp/sbcl/cl-change-case/cl-change-case-test.asd\"))" "--eval" "(asdf:test-system \"cl-change-case\")"
This is SBCL 2.1.0, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
WARNING: System definition file #P"/gnu/store/y3pflgxkfakhiljm4vmz3mxk34r38k7g-sbcl-cl-ppcre-unicode-2.1.1/share/common-lisp/sbcl/cl-ppcre-unicode/cl-ppcre-unicode.asd" contains definition for system "cl-ppcre-unicode-test". Please only define "cl-ppcre-unicode" and seconda
ry systems with a name starting with "cl-ppcre-unicode/" (e.g. "cl-ppcre-unicode/test") in that file.
;;; Computing Hangul syllable names
Running test suite CL-CHANGE-CASE
 Running test LOWER-CASE-STRINGS ...
 Running test LOWER-CASE-FIRST ...
 Running test STRING-LOWER-CASE-P ....
 Running test UPPER-CASE ....
 Running test UPPER-CASE-FIRST ...
 Running test STRING-UPPER-CASE-P ....
 Running test NO-CASE ..............................
 Running test CAMEL-CASE ..........
 Running test DOT-CASE .........
 Running test HEADER-CASE .........
 Running test SWAP-CASE ...
 Running test TITLE-CASE .........
 Running test OTHERS ......
 Did 97 checks.
    Pass: 97 (100%)
    Skip: 0 ( 0%)
    Fail: 0 ( 0%)

Unhandled ASDF/FIND-COMPONENT:MISSING-DEPENDENCY in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                              {1001878103}>:
  Component "cl-change-case-test" not found, required by
  #<SYSTEM "cl-change-case-test">

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1001878103}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK Component "cl-change-case-test" not found, required by #<SYSTEM "cl-change-case-test"> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* Component "cl-change-case-test" not found, required by #<SYSTEM "cl-change-case-test">)
2: (INVOKE-DEBUGGER Component "cl-change-case-test" not found, required by #<SYSTEM "cl-change-case-test">)
3: (ERROR ASDF/FIND-COMPONENT:MISSING-DEPENDENCY :REQUIRED-BY #<ASDF/SYSTEM:SYSTEM "cl-change-case-test"> :REQUIRES "cl-change-case-test")
4: (ASDF/FIND-COMPONENT:RESOLVE-DEPENDENCY-NAME #<ASDF/SYSTEM:SYSTEM "cl-change-case-test"> "cl-change-case-test" NIL)
5: (ASDF/PLAN:MAP-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case-test"> #<FUNCTION (LAMBDA (ASDF/PLAN::DEP-O ASDF/PLAN::DEP-C) :IN ASDF/PLAN:REDUCE-DIRECT-DEPENDENCIES) {100B4A132B}>)
6: (ASDF/PLAN:REDUCE-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case-test"> #<FUNCTION (LAMBDA (DO ASDF/PLAN::DC ASDF/PLAN::STATUS) :IN ASDF/ACTION:COMPUTE-ACTION-STAMP) {100B4A12DB}> #<ASDF/PLAN:ACTION-STATUS :BITS 4 :STAMP 3822415703
:LEVEL 0 :INDEX NIL>)
7: ((:METHOD ASDF/ACTION:COMPUTE-ACTION-STAMP (T ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) NIL #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case-test"> :JUST-DONE T) [fast-method]
8: ((:METHOD ASDF/ACTION:MARK-OPERATION-DONE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case-test">) [fast-method]
9: ((SB-PCL::EMF ASDF/ACTION:PERFORM) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case-test">)
10: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
11: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS :AROUND (T T)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case-test">) [fast-method]
12: ((:METHOD ASDF/PLAN:PERFORM-PLAN (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {1005A91713}>) [fast-method]
13: ((FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
14: ((:METHOD ASDF/PLAN:PERFORM-PLAN :AROUND (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {1005A91713}>) [fast-method]
15: ((:METHOD ASDF/OPERATE:OPERATE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case"> :PLAN-CLASS NIL :PLAN-OPTIONS NIL) [fast-method]
16: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case">)
17: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
18: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "cl-change-case">) [fast-method]
19: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> ASDF/LISP-ACTION:TEST-OP "cl-change-case")
20: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
21: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:TEST-OP "cl-change-case") [fast-method]
22: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {1005A87B4B}> :OVERRIDE T :KEY NIL :OVERRIDE-CACHE T :OVERRIDE-FORCING NIL)
23: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
24: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {1005A8521B}> :OVERRIDE NIL :KEY NIL :OVERRIDE-CACHE NIL :OVERRIDE-FORCING NIL)
25: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:TEST-OP "cl-change-case") [fast-method]
26: (ASDF/OPERATE:TEST-SYSTEM "cl-change-case")
27: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ASDF/OPERATE:TEST-SYSTEM "cl-change-case") #<NULL-LEXENV>)
28: (EVAL (ASDF/OPERATE:TEST-SYSTEM "cl-change-case"))
29: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(require :asdf)") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/96w6s8fk2y41yl6amkbr8h3z4d3av1dg-sbcl-cl-change-case-0.1.0-1.5ceff2a/share/common-lisp/sbcl/cl-change-case/cl-change-case.asd\"))") (:EVAL . "(asdf:load-asd
(truename \"/gnu/store/96w6s8fk2y41yl6amkbr8h3z4d3av1dg-sbcl-cl-change-case-0.1.0-1.5ceff2a/share/common-lisp/sbcl/cl-change-case/cl-change-case-test.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/96w6s8fk2y41yl6amkbr8h3z4d3av1dg-sbcl-cl-change-case-0.1.0-1.5ceff
2a/share/common-lisp/sbcl/cl-change-case/cl-change-case-test.asd\"))") (:EVAL . "(asdf:test-system \"cl-change-case\")") (:QUIT)))
30: (SB-IMPL::TOPLEVEL-INIT)
31: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
32: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-IMPL::START-LISP))
33: (SB-IMPL::START-LISP)```

**Comment by rudolfochrist** [comment]
Hi,
thanks for reporting this. 

You don't load any systems. You're just loading the ASD files, but this is different from loading the actual systems. You have to configure asdf in a way it can find the system sources or you have to place the sources where asdf can find them. The quick and dirty solution is to push the path to `asdf:*central-registry*`.

This works for me:
```
sbcl --non-interactive \
--no-userinit \
--eval '(require "asdf")' \
--eval '(push *default-pathname-defaults* asdf:*central-registry*)' \
--eval '(asdf:load-system "cl-change-case-test")' \
--eval '(asdf:test-system "cl-change-case")' 
```
This loads the `cl-change-case-test` system, which loads the `cl-change-case` system in return, because it holds a dependency on it. But unfortunately this results in another error:

```
sbcl --non-interactive \
--no-userinit \
--eval '(require "asdf")' \
--eval '(push *default-pathname-defaults* asdf:*central-registry*)' \
--eval '(asdf:load-system "cl-change-case-test")' \
--eval '(asdf:test-system "cl-change-case")'

This is SBCL 2.1.1, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
WARNING: System definition file #P"/Users/lispm/.lsd/client/cl-ppcre-2.1.1/cl-ppcre.asd" contains definition for system "cl-ppcre-test". Please only define "cl-ppcre" and secondary systems with a name starting with "cl-ppcre/" (e.g. "cl-ppcre/test") in that file.
WARNING: System definition file #P"/Users/lispm/.lsd/client/cl-ppcre-2.1.1/cl-ppcre-unicode.asd" contains definition for system "cl-ppcre-unicode-test". Please only define "cl-ppcre-unicode" and secondary systems with a name starting with "cl-ppcre-unicode/" (e.g. "cl-ppcre-unicode/test") in that file.
;;; Computing Hangul syllable names
Running test suite CL-CHANGE-CASE
 Running test LOWER-CASE-STRINGS ...
 Running test LOWER-CASE-FIRST ...
 Running test STRING-LOWER-CASE-P ....
 Running test UPPER-CASE ....
 Running test UPPER-CASE-FIRST ...
 Running test STRING-UPPER-CASE-P ....
 Running test NO-CASE ..............................
 Running test CAMEL-CASE ..........
 Running test DOT-CASE .........
 Running test HEADER-CASE .........
 Running test SWAP-CASE ...
 Running test TITLE-CASE .........
 Running test OTHERS ......
 Did 97 checks.
    Pass: 97 (100%)
    Skip: 0 ( 0%)
    Fail: 0 ( 0%)

WARNING:
   Computing just-done stamp in plan NIL for action (ASDF/LISP-ACTION:TEST-OP
                                                     "cl-change-case"), but dependency (ASDF/LISP-ACTION:TEST-OP
                                                                                        "cl-change-case-test") wasn't done yet!
```
I push some changes later to address this and further streamline the whole system definition.

**Comment by rudolfochrist** [comment]
I just released a new version (0.2.0) that should make it simpler:

```
sbcl --non-interactive \
--no-userinit \
--eval '(require "asdf")' \
--eval '(push *default-pathname-defaults* asdf:*central-registry*)' \
--eval '(asdf:load-system "cl-change-case")' \
--eval '(asdf:test-system "cl-change-case")'
This is SBCL 2.1.1, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
WARNING: System definition file #P"/Users/lispm/.lsd/client/cl-ppcre-2.1.1/cl-ppcre.asd" contains definition for system "cl-ppcre-test". Please only define "cl-ppcre" and secondary systems with a name starting with "cl-ppcre/" (e.g. "cl-ppcre/test") in that file.
WARNING: System definition file #P"/Users/lispm/.lsd/client/cl-ppcre-2.1.1/cl-ppcre-unicode.asd" contains definition for system "cl-ppcre-unicode-test". Please only define "cl-ppcre-unicode" and secondary systems with a name starting with "cl-ppcre-unicode/" (e.g. "cl-ppcre-unicode/test") in that file.
;;; Computing Hangul syllable names
Running test suite CL-CHANGE-CASE
 Running test LOWER-CASE-STRINGS ...
 Running test LOWER-CASE-FIRST ...
 Running test STRING-LOWER-CASE-P ....
 Running test UPPER-CASE ....
 Running test UPPER-CASE-FIRST ...
 Running test STRING-UPPER-CASE-P ....
 Running test NO-CASE ..............................
 Running test CAMEL-CASE ..........
 Running test DOT-CASE .........
 Running test HEADER-CASE .........
 Running test SWAP-CASE ...
 Running test TITLE-CASE .........
 Running test OTHERS ......
 Did 97 checks.
    Pass: 97 (100%)
    Skip: 0 ( 0%)
    Fail: 0 ( 0%)
```

**Comment by rudolfochrist** [comment]
resolved in c976d7e65db314844284dbc9c4c52ee4a320d6e2
